Sources/Epoxy/Core/ButtonRow.swift

```diff
@@ -0,0 +1,203 @@
+        }
+    }
+
+    public convenience init(style: Style) {
+        self.init(type: style.type)
+
+        translatesAutoresizingMaskIntoConstraints = false
+
+        tintColor = style.tintColor
+
+        if let titleStyle = style.titleStyle {
+            titleLabel?.do {
+                $0.font = titleStyle.font
+                $0.textAlignment = titleStyle.alignment
+                $0.numberOfLines = titleStyle.numberOfLines
+                $0.lineBreakMode = titleStyle.lineBreakMode
+            }
+        }
+
+        addTarget(self, action: #selector(buttonDidTouchDown(_:)), for: .touchDown)
+        addTarget(self, action: #selector(buttonDidClick(_:)), for: .touchUpInside)
+    }
+}
+
+// MARK: ContentConfigurableView
+
+extension ButtonRow: ContentConfigurableView {
+    public enum Content: Equatable {
+        public struct StateContent: Equatable {
+            public let image: ImageRow.ImageType?
+            public let title: TextRow.Content?
+            public let titleColor: UIColor
+
+            public init(
+                image: ImageRow.ImageType? = nil,
+                title: TextRow.Content? = nil,
+                titleColor: ConvertibleToColor = UIColor.label
+            ) {
+                self.image = image
+                self.title = title
+                self.titleColor = titleColor.color
+            }
+        }
+
+        case normal(StateContent)
+        case disabled(StateContent)
+        case selected(StateContent)
+        case highlighted(StateContent)
+    }
+
+    public func setContent(_ content: Content, animated _: Bool) {
+        func _set(with stateContent: Content.StateContent, for state: UIControl.State) {
+            if let image = stateContent.image {
+                setImage(image.image, for: state)
+            }
+
+            if let title = stateContent.title {
+                switch title {
+                case .text(let value):
+                    setTitle(value, for: state)
+
+                case .attributedText(let value):
+                    setAttributedTitle(value, for: state)
+                }
+            }
+
+            setTitleColor(stateContent.titleColor, for: state)
+        }
+
+        switch content {
+        case .normal(let stateContent):
+            isEnabled = true
+            isSelected = false
+            isHighlighted = false
+
+            _set(with: stateContent, for: .normal)
+
+        case .disabled(let stateContent):
+            isEnabled = false
+            isSelected = false
+            isHighlighted = false
+
+            _set(with: stateContent, for: .disabled)
+
+        case .selected(let stateContent):
+            isSelected = true
+            isHighlighted = false
+
+            _set(with: stateContent, for: .selected)
+
+        case .highlighted(let stateContent):
+            isSelected = false
+            isHighlighted = true
+
+            _set(with: stateContent, for: .highlighted)
```

Comment:
When setting `.selected` and `.highlighted`, `isEnabled` is not modified. This is debatable because I'm not sure what the connection is between these three states.